src/components/HomePage/SearchResultsArea/Offer/OfferContent.spec.js

```diff
@@ -30,32 +56,497 @@ describe("OfferContent", () => {
+
+        it ("should hide offer if the owner company clicks the hide button", async () => {
+
+            const anotherOffer = new Offer({
+                id: "id1",
+                title: "position1",
+                owner: "company_id",
+                ownerName: "company1",
+                ownerLogo: "",
+                location: "location1",
+                jobStartDate: (new Date()).toISOString(),
+                publishDate: "2021-04-22T22:35:57.177Z",
+                publishEndDate: "2021-09-19T23:00:00.000Z",
+                description: "description1",
+                isHidden: false,
+                hiddenReason: null,
+                adminReason: null,
+            });
+
+            // Simulate request success
+            fetch.mockResponse(JSON.stringify({ mockData: true }));
+
+            useSession.mockImplementation(() => ({
+                isLoggedIn: true, data: { company: { name: "company1", _id: "company_id" } },
+            }));
+
+            renderWithStoreAndTheme(
+                <OfferContent offer={anotherOffer} />,
+                { theme }
+            );
+
+            expect(screen.queryByText(
+                "This offer is hidden so it won't show up in search results",
+                { exact: false }
+            )).not.toBeInTheDocument();
+
+            const visibilityButton = screen.getByRole("hideEnableOfferButton");
+            expect(visibilityButton).toBeInTheDocument();
+
+            await act(async () => {
+                await fireEvent.click(visibilityButton);
+            });
+
+            expect(screen.queryByText(
+                "This offer is hidden so it won't show up in search results", { exact: false }
+            )).toBeInTheDocument();
+        });
+
+        it ("should hide offer if an admin clicks the hide button", async () => {
+
+            const anotherOffer = new Offer({
+                id: "id1",
+                title: "position1",
+                owner: "company_id",
+                ownerName: "company1",
+                ownerLogo: "",
+                location: "location1",
+                jobStartDate: (new Date()).toISOString(),
+                publishDate: "2021-04-22T22:35:57.177Z",
+                publishEndDate: "2021-09-19T23:00:00.000Z",
+                description: "description1",
+                isHidden: false,
+                hiddenReason: null,
+                adminReason: null,
+            });
+
+            // Simulate request success
+            fetch.mockResponse(JSON.stringify({ mockData: true }));
+
+            useSession.mockImplementation(() => ({ isLoggedIn: true, data: { email: "admin@admin.com", isAdmin: true } }));
+
+            renderWithStoreAndTheme(
+                <OfferContent offer={anotherOffer} />,
+                { theme }
+            );
+
+            expect(screen.queryByText(
+                "This offer is hidden so it won't show up in search results",
+                { exact: false }
```

Comment:
why do you need the exact:false? What is the exact text?

Reply:
@imnotteixeira  In this case, if I didn't use the `exact: false`, and for example, the text on the screen was `This offer is hidden so it won't show up in search results.` (with a dot), the test would pass, even though the text `This offer is hidden so it won't show up in search results` was on the screen. If I use the `exact: false`, this problem does not exist. With that said, I think it's a good practice to use the `exact: false` when we are testing if something is not on the screen.

Reply:
https://testing-library.com/docs/queries/about/#textmatch-examples